EIPS/eip-5164.md

```diff
@@ -0,0 +1,211 @@
+---
+eip: 5164
+title: Cross-Chain Execution
+description: This specification defines an interface that supports execution across EVM networks.
+author: Brendan Asselstine (@asselstine), Anna Carroll (@anna-carroll), Hadrien Croubois (@Amxx), Nam Chu Hoai (@nambrot), Georgios (@geogons), Theo Gonella (@mintcloud), Rafael Solari (@rsolari), Auryn Macmillan (@auryn-macmillan), Nathan Ginnever (@nginnever)
+discussions-to: https://ethereum-magicians.org/t/eip-5164-cross-chain-execution/9658
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-06-14
+---
+
+## Abstract
+
+This specification defines a cross-chain execution interface for EVM-based blockchains. Developers will be able to have contracts call contracts on other chains. There are two parts to this spec: the "sending" interface and the "receiving" interface. The "sending" interface specifies the interface to send a call to another chain; this interface must be implemented by bridges. The "receiving" interface must be implemented by the developer; it is an interface for receiving cross-chain calls. This specification is agnostic of the bridge implementation; it simply standardizes the call interfaces.
+
+## Motivation
+
+Many Ethereum protocols need to coordinate state changes across multiple EVM-based blockchains. These chains often have native or third-party bridges that allow Ethereum contracts to execute code. However, bridges have different APIs so bridge integrations are bespoke.
+
+Bridge technology is improving rapidly, and there are multiple bridges to choose from. Each one affords different properties; with varying degrees of security, speed, and control.
+
+By standardizing a cross-chain execution interface, we can cleanly separate the transport layer from the application layer. Cross-chain execution becomes another composable piece with which the Ethereum ecosystem can build shared infrastructure.
+
+## Specification
+
+This specification allows contracts on one chain to send messages to contracts on another chain. There are two key contracts:
```

Comment:
Looks like `CrossChainRelayer` and `CrossChainReceiver` are interfaces, not contracts?

```suggestion
This specification allows contracts on one chain to send messages to contracts on another chain. There are two interfaces need to be implemented:
```

Reply:
Fixed in this commit: https://github.com/ethereum/EIPs/pull/5164/commits/772f1c7b03c55496f2f724073e8ebb227e154a5c#diff-41efe0617f861cc7330b025ecc86d55fe777d3c503208d83dca04258de4f5f41R29